File: articles/childhood-development-in-jungian-and-post-jungian-thought/go/developmental_coherence.go
package main

import "fmt"

func developmentalStrain(egoDifferentiation, relationalSecurity, symbolicCapacity, complexActivation float64) float64 {
	eR := egoDifferentiation - relationalSecurity
	sR := symbolicCapacity - relationalSecurity
	return eR*eR + sR*sR + complexActivation*complexActivation
}

func developmentalCoherence(
	egoDifferentiation,
	relationalSecurity,
	symbolicCapacity,
	affectRegulation,
	symbolicPlay,
	complexActivation,
	strain float64,
) float64 {
	return 0.55*egoDifferentiation +
		0.70*relationalSecurity +
		0.60*symbolicCapacity +
		0.48*affectRegulation +
		0.34*symbolicPlay -
		0.50*complexActivation -
		0.24*strain
}

func main() {
	strain := developmentalStrain(0.82, 0.90, 0.90, 0.18)
	score := developmentalCoherence(0.82, 0.90, 0.90, 0.82, 0.90, 0.18, strain)

	fmt.Printf("Synthetic developmental strain: %.3f\n", strain)
	fmt.Printf("Synthetic developmental coherence score: %.3f\n", score)
}

File: articles/childhood-development-in-jungian-and-post-jungian-thought/fortran/developmental_coherence.f90
! Childhood Development in Jungian and Post-Jungian Thought
! Fortran example: simple synthetic developmental coherence score

program developmental_coherence_demo
  implicit none

  real :: strain
  real :: score

  strain = (0.82 - 0.90)**2 + (0.90 - 0.90)**2 + 0.18**2

  score = 0.55 * 0.82 + 0.70 * 0.90 + 0.60 * 0.90 + &
          0.48 * 0.82 + 0.34 * 0.90 - 0.50 * 0.18 - &
          0.24 * strain

  print *, "Synthetic developmental strain:", strain
  print *, "Synthetic developmental coherence score:", score
end program developmental_coherence_demo
